### PR TITLE
Update to Rails 5.0

### PR DESCRIPTION
**Pull request by MrSerth**
This PR includes many changes all are necessary changes to update mammooc to Rails 5. 

I tried to test everything manually and with our test suite, I truly believe nothing should break. In order to verify that on a deployed system, I would like to do some further testing on our Heroku dev app once it's merged prior to a master deployment.

(And yeah: The first push got a green build status!)

### CODE REVIEW

File: app/controllers/evaluations_controller.rb [hunk 1]
```diff
@@ -32,68 +32,68 @@ def process_feedback
   def export_overall_course_rating
     if params[:provider].present? && params[:course_id].present?
       mooc_provider = MoocProvider.find_by!(name: params[:provider])
-      course = [Course.find_by!(provider_course_id: params[:course_id], mooc_provider: mooc_provider)].first()
+      course = [Course.find_by!(provider_course_id: params[:course_id], mooc_provider: mooc_provider)].first
```
Comment (rosabraatz) [inline app/controllers/evaluations_controller.rb]: Why don't use: 'Course.find_by!(provider_course_id: params[:course_id], mooc_provider: mooc_provider)' ? This should return exactly one course (the first one in the database matching the criteria).

File: app/controllers/evaluations_controller.rb [hunk 2]
```diff
@@ -32,68 +32,68 @@ def process_feedback
...
+    respond_to do |format|
+      format.json { render json: {error: e.message}, status: :recordNotFound }
+    end
   end
 
   def export_course_evaluations
     if params[:provider].present? && params[:course_id].present?
       mooc_provider = MoocProvider.find_by!(name: params[:provider])
-      course = [Course.find_by!(provider_course_id: params[:course_id], mooc_provider: mooc_provider)].first()
+      course = [Course.find_by!(provider_course_id: params[:course_id], mooc_provider: mooc_provider)].first
```
Comment (rosabraatz) [inline app/controllers/evaluations_controller.rb]: See comment above.

File: app/controllers/evaluations_controller.rb (hunk 1 above)
Comment (MrSerth) [inline app/controllers/evaluations_controller.rb]: @Atyanar Please have a look at this code, it was part of your Pull Request #696.

File: app/controllers/evaluations_controller.rb (hunk 2 above)
Comment (MrSerth) [inline app/controllers/evaluations_controller.rb]: @Atyanar Same as above :)

File: config/application.rb [hunk 3]
```diff
@@ -50,24 +47,5 @@ class Application < Rails::Application
 
     # Force SSL for all connections in single-mode
     config.force_ssl = true if ENV['FORCE_SSL'] == 'true'
-
-    config.middleware.insert_before 0, 'Rack::Cors' do
```
Comment (rosabraatz) [inline config/application.rb]: Why did you delete all the things with CORS?

Comment (MrSerth) [inline config/application.rb]: They aren't deleted but moved to a new file `/config/initializers/cors.rb` which is the default in Rails 5.

File: config/initializers/new_framework_defaults.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+# Be sure to restart your server when you modify this file.
+#
+# This file contains migration options to ease your Rails 5.0 upgrade.
+#
+# Once upgraded flip defaults one by one to migrate to the new default.
+#
+# Read the Rails 5.0 release notes for more info on each option.
+
+# Enable per-form CSRF tokens. Previous versions had false.
+Rails.application.config.action_controller.per_form_csrf_tokens = true
```
Comment (rosabraatz) [inline config/initializers/new_framework_defaults.rb]: This just looks like a typo: 'per_form'. If it is correct, forget about my comment.

File: app/controllers/evaluations_controller.rb (hunk 1 above)
Comment (MrSerth) [inline app/controllers/evaluations_controller.rb]: I agree to @rosabraatz' comment as find_by! always limits the SQL query to return one result. Why should we take one result, put it in an array (wich then will always have a size of zero or one) and get the first element (resulting in `nil` or the result from find_by!). Do you, @Atyanar, agree if I change the code here and in the method below?

File: config/initializers/new_framework_defaults.rb (hunk 4 above)
Comment (MrSerth) [inline config/initializers/new_framework_defaults.rb]: Thanks for reading that careful (I might have overlooked that in another PR), I just wondered myself and double-checked that. According to the Guid on the Ruby on Rails website [1], this is correct (and it doesn't throw an error or exception)

[1]: http://edgeguides.rubyonrails.org/configuring.html#configuring-action-controller

File: spec/models/course_spec.rb [hunk 5]
```diff
@@ -25,8 +25,8 @@
     let!(:course1) do
       FactoryGirl.create(:course,
         mooc_provider_id: provider.id,
-        start_date: Time.zone.local(2015, 03, 15),
-        end_date: Time.zone.local(2015, 03, 17),
+        start_date: Time.zone.local(2015, 0o3, 15),
```
Comment (rosabraatz) [inline spec/models/course_spec.rb]: Could you explain this syntax '0o3'?

File: config/application.rb (hunk 3 above)
Comment (rosabraatz) [inline config/application.rb]: Sorry, I didn't read until the end. ;)

**Comment by rosabraatz** [comment]
I think everything looks good. It was pretty much to read, I hope I didn't overlook anything. ;)

File: spec/models/course_spec.rb (hunk 5 above)
Comment (MrSerth) [inline spec/models/course_spec.rb]: Its an octal representation (as it was before as well) but more explicit. I'll refactor that to remove the leading `0o`.

File: app/controllers/evaluations_controller.rb (hunk 1 above)
Comment (Atyanar) [inline app/controllers/evaluations_controller.rb]: Yes, I completely agree! (Though I just copied this part over from the original export function).

**Comment by MrSerth** [comment]
I included all changes and - finally - all specs run _on CircleCI as well_ (I still don't know what's different there, it was a hell of "bug fixing" to get that solved!)

Here's a comparison between changes from yesterday to today. It might be faster to read only the following diff: https://github.com/mammooc/mammooc.org/compare/9d91253...updates
